src/js/actions/layers.js

```diff
@@ -1377,45 +1382,153 @@ define(function (require, exports) {
     };
 
     /**
+     * Helper function to find the right most artboard. It assumes a non-empty list of artboards. 
+     * 
+     * @private 
+     * @param {Immutable.List.<Layer>} artboards
+     * @return {Layer}
+     */
+    var _findRightMostArtboard = function (artboards) {
+        var layer = artboards.reduce(function (selectedLayer, currentLayer) {
+            if (currentLayer.bounds.right > selectedLayer.bounds.right) {
+                return currentLayer;
+            } else {
+                return selectedLayer;
+            }
+        }, artboards.first());
+        return layer;
+    };
+
+    /**
+     * Helper function to get Bounds from specs (template preset)
+     *
+     * @private
+     * @param {object} specs 
+     * @param {Immutable.List.<Layer>} artboards
+     * @return {{finalBounds: Bounds, changeLayerRef: Boolean}}
+     */
+    var _getBoundsFromTemplate = function (specs, artboards) {
+        var preset = specs.preset,
+            index = _.findIndex(templates, function (obj) {
+                return obj.preset === preset;
+            });
+
+        var height = templates[index].height,
+            width = templates[index].width,
+            changeLayerRef = false,
+            finalBounds;
+
+        // if artboards are empty, insert template at first position 
+        if (artboards.isEmpty()) {
+            changeLayerRef = true;
+            finalBounds = new Bounds({
+                top: DEFAULT_ARTBOARD_BOUNDS.top,
+                bottom: height,
+                left: DEFAULT_ARTBOARD_BOUNDS.left,
+                right: width
+            });
+        } else {
+            // find the rightmost artboard and insert the template after this 
+            var rightMostLayer = _findRightMostArtboard(artboards);
+            var offset = 100;
+            finalBounds = rightMostLayer.bounds.merge({
+                bottom: rightMostLayer.bounds.top + height,
+                left: rightMostLayer.bounds.left + offset + rightMostLayer.bounds.width,
+                right: rightMostLayer.bounds.right + offset + width
+            });
+        }
+
+        return {
+            finalBounds: finalBounds,
+            changeLayerRef: changeLayerRef
+        };
+    };
+
+    /**
+     * Helper function to get the bounds of an artboard with no prior input
+     * 
+     * @private
+     * @param {Immutable.List.<Layer>} artboards
+     * @param {Immutable.List.<Layer>} selectedArtboards 
+     * @return {{finalBounds: Bounds, changeLayerRef: Boolean}}
```

Comment:
Basic types are expressed in lower case: `object`, `string`, `number`, `boolean`, `undefined`, etc.